### PR TITLE
[Core] Report disabled usage stats

### LINKED ISSUE #33486: [core][python3.11] Fix CI tests
### What happened + What you expected to happen

- [ ] python/ray/tests/test_usage_stats.py
- [ ] python/ray/tests/test_runtime_env_complicated.py
- [ ] python/ray/tests/test_runtime_env.py

### Versions / Dependencies

python 3.11
linux

### Reproduction script

na

### Issue Severity

None

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
When the usage stats is disabled, we send a message to the server indicating there is a cluster with usage stats disabled.

## Related issue number
Closes #33486

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/usage/usage_lib.py [hunk 1]
```diff
@@ -787,6 +797,19 @@ def get_cluster_metadata(gcs_client) -> dict:
...
 
+def generate_disabled_report_data() -> UsageStatsToReport:
+    """Generate the report data indicating usage stats is disabled"""
+    data = UsageStatsToReport(
+        schema_version=usage_constant.SCHEMA_VERSION,
+        source=os.getenv(
+            usage_constant.USAGE_STATS_SOURCE_ENV_VAR,
+            usage_constant.USAGE_STATS_SOURCE_OSS,
+        ),
+        collect_timestamp_ms=int(time.time() * 1000),
```
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:808]: Those are the only data we send to the server when usage stats is disabled.

File: doc/source/cluster/usage-stats.rst [hunk 2]
```diff
@@ -41,6 +41,7 @@ There are multiple ways to disable usage stats collection before starting a clus
 
 Currently there is no way to enable or disable collection for a running cluster; you have to stop and restart the cluster.
 
+If usage stats is disabled, Ray will [PlaceHolder].
```
Comment (jjyao) [inline doc/source/cluster/usage-stats.rst:44]: @richardliaw can you help with the doc?

Comment (richardliaw) [reply doc/source/cluster/usage-stats.rst:44]: I think we don't need to add anything

File: python/ray/_private/usage/usage_lib.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/_private/usage/usage_lib.py:808]: so only the timestamp?

File: doc/source/cluster/usage-stats.rst [hunk 3]
```diff
@@ -41,7 +41,6 @@ There are multiple ways to disable usage stats collection before starting a clus
 
 Currently there is no way to enable or disable collection for a running cluster; you have to stop and restart the cluster.
 
-
 How does it work?
```
Comment (rkooo567) [inline doc/source/cluster/usage-stats.rst:44]: I think we should update the doc to explain this?

File: dashboard/modules/usage_stats/usage_stats_head.py [hunk 4]
```diff
@@ -166,6 +184,8 @@ async def run(self, server):
         )
         if not self.usage_stats_enabled:
             logger.info("Usage reporting is disabled.")
+            if not ray_usage_lib.is_usage_stats_ray_init_cluster():
```
Comment (rkooo567) [inline dashboard/modules/usage_stats/usage_stats_head.py:187]: what's this condition mean exactly?

File: python/ray/_private/usage/usage_lib.py [hunk 5]
```diff
@@ -494,6 +492,18 @@ def set_usage_stats_enabled_via_env_var(enabled) -> None:
     os.environ[usage_constant.USAGE_STATS_ENABLED_ENV_VAR] = "1" if enabled else "0"
 
 
+def set_usage_stats_ray_init_cluster() -> None:
```
Comment (rkooo567) [inline python/ray/_private/usage/usage_lib.py:495]: this seems a bit hacky.. is there any way we just figure this out from existing path?

Comment (rkooo567) [reply python/ray/_private/usage/usage_lib.py:495]: Or alternatively, we just pass a flag when we start a dashboard instead?

File: python/ray/tests/test_usage_stats.py [hunk 6]
```diff
@@ -1290,7 +1259,47 @@ def test_first_usage_report_delayed(monkeypatch, ray_start_cluster, reset_usage_
         assert (session_path / usage_constants.USAGE_STATS_FILE).exists()
 
 
-def test_usage_report_disabled(monkeypatch, ray_start_cluster, reset_usage_stats):
+def test_usage_report_disabled_ray_init_cluster(
+    monkeypatch, start_usage_stats_server, reset_usage_stats, shutdown_only
+):
+    """
+    Make sure we don't send anything to the server for the ray.init cluster
```
Comment (rkooo567) [inline python/ray/tests/test_usage_stats.py:1266]: why is that?

**Review by rkooo567 — APPROVED** [review]
LGTM. Minor comment for how we send information regarding if it is ray.init cluster to the dashboard.

File: python/ray/_private/usage/usage_lib.py (hunk 1 above)
Comment (jjyao) [reply python/ray/_private/usage/usage_lib.py:811]: cc @richardliaw @robertnishihara This is probably what you care about.

**Review by richardliaw — APPROVED** [review]

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**